### PR TITLE
Add tests for construction

### PR DESCRIPTION
**Pull request by Twon**

### CODE REVIEW

**Comment by Twon** [comment]
The current Windows failure is due to the Visual Studio implementation of unique_ptr not setting the pointer to null in the destructor.  So when the test explicitly calls the destructor it then fails when the destructor is again implicitly called as the end of the block's scope.

The test needs to be refactored to not explicitly call the destructor, but to instead use the scoping to invoke the destructor and then check the deleter count.  I shall look at this tomorrow.

File: indirect.h [hunk 1]
```diff
@@ -15,7 +15,8 @@ struct default_copy {
 template <class T, class C = default_copy<T>, bool CanBeEmptyBaseClass = std::is_empty_v<C> && !std::is_final_v<C> >
 class indirect_base {
 protected:
-  indirect_base() noexcept(noexcept(C())) = default;
+  template<class U = C, class = std::enable_if_t<std::is_default_constructible_v<U>>>
+  indirect_base() noexcept(noexcept(C())) {}
```
Comment (jbcoe) [inline indirect.h]: I think we should just require that C is noexcept default constructible. 
As a reference Impl I (weakly) favour simplicity over premature generality.

Comment (Twon) [reply indirect.h]: Unfortunately, Lambdas are not default-constructible, so the enable_if was required to support disabling the default-constructor when containing Lambdas.  

Happy to add a check for is no except default constructible to the enable if you think that helps?  However, I think it takes less effort not to have to even think about this issue and have the class flexible enough to enable the noexecept-construct based on if the underlying class does support it.

Comment (jbcoe) [reply indirect.h]: I'm happy to ignore C,D as lambdas as use cases. I'd like to keep this as simple as possible so folk feel comfortable using it. Every line of meta-programming hurts us I fear.

Comment (Twon) [reply indirect.h]: Ok, well if you don't mind I'd like to submit as is?  I'll then follow up with a pull request explicitly disabling support for lambda and adding static asserts to provide useful error diagnostics if they are used with lambdas.

**Pull request merged**